gmso/formats/itp.py

```diff
@@ -0,0 +1,331 @@
+            site_4.atom_type.name,
+        ),
+        member_classes=(
+            site_1.atom_type.atomclass,
+            site_2.atom_type.atomclass,
+            site_3.atom_type.atomclass,
+            site_4.atom_type.atomclass,
+        ),
+        parameters={
+            "c0": c0 * u.Unit("kJ / (deg**2)"),
+            "c1": c1 * u.Unit("kJ / (deg**2)"),
+            "c2": c2 * u.Unit("kJ / (deg**2)"),
+            "c3": c3 * u.Unit("kJ / (deg**2)"),
+            "c4": c4 * u.Unit("kJ / (deg**2)"),
+            "c5": c5 * u.Unit("kJ / (deg**2)"),
+        },
+    )
+    pdb.add_connection(diehdrals)
+
+
+def _parse_harmonic(pdb, line):
+    dind_1, dind_2, dind_3, dind_4 = (
+        int(line.split()[0]) - 1,
+        int(line.split()[1]) - 1,
+        int(line.split()[2]) - 1,
+        int(line.split()[3]) - 1,
+    )
+    d_K, d_phi, d_n = (
+        float(line.split()[5]),
+        float(line.split()[6]),
+        float(line.split()[7]),
+    )
+
+    site_1 = pdb.sites[dind_1]
+    site_2 = pdb.sites[dind_2]
+    site_3 = pdb.sites[dind_3]
+    site_4 = pdb.sites[dind_4]
+    diehdrals = Dihedral(connection_members=(site_1, site_2, site_3, site_4))
+    K = d_K
+    phi_eq = d_phi
+    n = d_n
+    diehdrals.dihedral_type = DihedralType(
+        name="HarmonicTorsionPotential",
+        expression="k * (1 + cos(n * phi - phi_eq))**2",
+        independent_variables={"phi"},
+        member_types=(
+            site_1.atom_type.name,
+            site_2.atom_type.name,
+            site_3.atom_type.name,
+            site_4.atom_type.name,
+        ),
+        member_classes=(
+            site_1.atom_type.atomclass,
+            site_2.atom_type.atomclass,
+            site_3.atom_type.atomclass,
+            site_4.atom_type.atomclass,
+        ),
+        parameters={
+            "k": K * u.Unit("kJ / (deg**2)"),
+            "phi_eq": phi_eq * u.deg,
+            "n": n * u.dimensionless,
+        },
+    )
+    pdb.add_connection(diehdrals)
+
+
+def _parse_dihedrals(file, pdb):
+    # dry run to get the dihedral type
+    for i, line in enumerate(file):
+        if "dihedrals" in line:
+            for i, line_1 in enumerate(file):
+                if line_1.split() and ";" not in line_1.split():
+                    d_type = int(line_1.split()[4])
+                    break
+            break
+
+    dicts_dtype = {
+        1: _parse_harmonic,
+        3: _parse_RB,
+    }  # This one should come from the GROMACS page
+    # https://manual.gromacs.org/documentation/current/reference-manual/topologies/topology-file-formats.html
+    parser = dicts_dtype[d_type]
+
+    # file.seek(0)
+    # #return to previous line command not working (will fix it later)
+
+    for line in file:
+        if line.split() and ";" not in line.split():
+            parser(pdb, line)
+        else:
+            break
+
+    return pdb
+
+    # https://github.com/mosdef-hub/gmso/tree/main/gmso/lib/jsons
+
+
+def read_itp(itp_file):
```

Comment:
Lets move this function to the top of this file since it will be the user-facing one.